src/compiler/checker.ts

```diff
@@ -7254,14 +7253,37 @@ module ts {
                 }
             }
 
+            function isImportedNameFromExternalModule(n: Node): boolean {

```

Comment:
it is both the question of style (use explicit inputs\outputs for function) and potential optimization opportunities (in theory nested function that does not any captured data can be lifted to a higher scope)